broker/redis.go

```
@@ -204,10 +209,11 @@ func (b *RedisBroker) publishOn(msg []byte) {
 	appendedVals, err := redis.Values(conn.Do("EXEC"))
 	if err != nil {
 		util.CountWithData("RedisBroker.publishOn.error", 1, "error=%s", err)
+		return err
 	}
 	appendedLen := appendedVals[0].(int64)
 
-	conn.Send("PUBLISH", b.channel.id(), appendedLen)
+	return conn.Send("PUBLISH", b.channel.id(), appendedLen)
```

We can probably keep the behavior and silently return nil for the error (even if there was one) just to satisfy the Writer interface.

If you think that's better, I'm :+1: (was mostly unsure regarding the behavior that we want)